spec.html

```diff
@@ -21446,24 +22027,46 @@ <h1>Runtime Semantics: ClassFieldDefinitionEvaluation</h1>
 
     <emu-clause id="sec-static-semantics-classelementevaluation" type="sdo" aoid="ClassElementEvaluation">
       <h1>Runtime Semantics: ClassElementEvaluation</h1>
-      <p>With parameters _object_ and _enumerable_.</p>
+      <p>With parameters _object_, _enumerable_, and _metadataList_.</p>
 
       <emu-grammar>
-        ClassElement : FieldDefinition `;`
+        ClassElement : DecoratorList? FieldDefinition `;`
+      </emu-grammar>
+      <emu-alg>
+        1. If |DecoratorList_opt| is present, then
+          1. Let _decorators_ be ? DecoratorListEvaluation of |DecoratorList|.
+        1. Else, let _decorators_ be a new empty List.
```

Comment:
I've tried to do this by creating a `MaybeDecoratorList` grammar, let me know if this solves the issue

Reply:
(I got a notification for the above comment, but I didn't remember the context after 7 months, so I clicked the link, but GitHub wouldn't show me this conversation because you'd already marked it resolved. I had to click a bunch of "Show resolved" before I found it. For this reason, and others, I think it's better to allow the *reviewer* to mark a conversation as resolved. I believe that's the local convention, though I couldn't find anything to that effect in tc39/how-we-work.)

Reply:
Sorry, part of my workflow in large PRs is usually to use these comments as a todo list, and to mark as resolved when I've checked this review off the list. I'll unresolve these.

As for this issue, `MaybeDecoratorList` didn't actually work because it'll fail to parse in a lot of situations, `DecoratorList` needs to be truly optional. I can't find an example of an SDO where the left hand side of the SDO itself is optional, and if the whole production doesn't exist it returns a different value, could use some advice on the best way to do this.

Reply:
I'll think about it more in my next review (if it hasn't already been resolved by then).